### DISCUSSION TITLE
Publishing with Pages: moving master to main

### DISCUSSION THREAD

**Discussion by denisecase** [community/community#23147] [Other Feature Feedback, Questions, & Ideas]
<p>Hi all,</p>
<p>We love GitHub Pages and have begun moving our <strong>master</strong> branch to <strong>main</strong> as recommended below:</p>
<ol>
<li><a href="https://www.hanselman.com/blog/EasilyRenameYourGitDefaultBranchFromMasterToMain.aspx" rel="nofollow noopener">Easily rename your Git default branch from master to main</a></li>
</ol>
<p>When will we be able to publish from “main” branch instead of “master” in GitHub Pages?</p>
<p>Thank you!<br>
Denise</p>

**Comment by otter-computer** [comment, marked as answer]
<p>We don’t currently have anything to announce on precisely when the transition from <strong>master</strong> to <strong>main</strong> will be happening. In the case of GitHub Pages it’s specifically coded that you need to deploy your site on the <strong>master</strong> branch. This will be changing when we do make these changes to the default branches but we don’t have an exact date yet. I’d recommend keeping an eye on the <a href="https://github.blog/" rel="nofollow noopener">GitHub Blog</a> for more news on when this will go live.</p>
<p>For now though you’ll still need to deploy your Pages site from the <strong>master</strong> or <strong>gh-pages</strong> branch, <a href="https://help.github.com/en/github/working-with-github-pages/about-github-pages#publishing-sources-for-github-pages-sites" rel="nofollow noopener">depending on the type of Pages site you’re publishing</a>.</p>

**Reply by niemasd** [reply]
Pinging this, as it seems to be the case ~3 years later. Any updates for deploying GitHub Pages from "main" (or whatever the default branch is) instead of "master"?

**Comment by denisecase** [comment]
<p>Will do - thanks, <a class="mention" href="/thomasshaped">@thomasshaped</a>.</p>
<p>A possible option would be for GitHub Pages publish to the <strong>default branch</strong>.  Selecting either gh-pages or the default branch would work for older sites using “master”, as well as new sites where the default is “main” (or any other option).</p>
<p>Looking forward to the changes!</p>

**Comment by QuintinWillison** [comment]
<p>Hello <a class="mention" href="/thomasshaped">@thomasshaped</a>,</p>
<p>Thanks for the initial response on this. It’s reassuring to hear that change is afoot.</p>
<p>Do you know if there’s been any progress on this?<br>
I can’t find any mention on the blog.</p>
<p>Thanks.</p>

**Comment by otter-computer** [comment]
<p><a class="mention" href="/quintinwillison">@QuintinWillison</a> Not quite yet, though work is progressing! There’s unfortunately a lot of legacy assumptions that need to be catered for when making a change like this so the work is a little more complicated than just changing some references to <code>master</code> in the original code. I can’t give you an exact date on when these changes will be coming but they are on track to happen very soon!</p>

**Comment by otter-computer** [comment]
<p>Wanted to drop an update in this thread that our repository tracking branch renaming has gone public:</p>
<aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/github/renaming" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2439407/181075335-97254f03-7721-4644-8c4d-33059fd3235d.png" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/github/renaming" target="_blank" rel="nofollow noopener">github/renaming</a></h3>

<p>Guidance for changing the default branch name for GitHub repositories - github/renaming</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>There’s no change for the status of Pages right now—you’ll need to keep using either <code>master</code> or <code>gh-pages</code> for the time being—but future updates will be posted in here and likely on the blog too. From the README it looks like we’re targeting later this year for the rollout of changes to Pages sites. I’d recommend you keep an eye on this repository for updates on this!</p>

**Comment by QuintinWillison** [comment]
<p>Thanks for keeping on top of this, <a class="mention" href="/thomasshaped">@thomasshaped</a>. :smile:</p>

**Comment by wmdietl** [comment]
<p>This thread is a bit old already, but seemed like the best place for my comment:</p>
<ul>
<li>I created a new <a href="http://orgname.github.io" rel="noopener nofollow ugc">orgname.github.io</a> repository.</li>
<li>I go to the site and get “404 There isn’t a GitHub Pages site here.”</li>
<li>I waited a bit, updated the repo, but nothing changed.</li>
<li>I finally realized that in “<a href="http://orgname.github.io" rel="noopener nofollow ugc">orgname.github.io</a> -&gt; Settings -&gt; Options -&gt; GitHub Pages” under “Source” it said: “Your GitHub Pages site is currently being built from the master branch.”</li>
<li>As the default for new repositories is now “main”, GitHub Pages didn’t find a “master” branch and gives the 404 error.</li>
<li>Once I changed the source to “main” the site was available.</li>
</ul>
<p>It would be nice if GitHub Pages also used “main” as the default to match the new repository default.</p>

**Comment by Khairunj** [comment]
<p>Hello Thomasshaped</p>
<p>I am the Program manager in https//docs.microsoft.com team and we have hundreds of repo with “master” as the default branch and we are waiting to know when GitHub will provide the tool to help with renaming the master to main as mentioned at <a href="https://github.com/github/renaming#later-this-year-seamless-move-for-existing-repositories-" rel="noopener nofollow ugc">https://github.com/github/renaming#later-this-year-seamless-move-for-existing-repositories-</a></p>

**Comment by OnlineTeachingNow** [comment]
<p>Hello, in reading the renaming guidance you linked, I am under the impression that main has all the functionality of ‘master’. However, I am unable to push to main and can only push to a master folder that I create. When I push to main, I get the error: src refspec main does not match any<br>
error: failed to push some refs to ‘<a href="https://github.com/blah" rel="noopener nofollow ugc">BLAH · GitHub</a> blah blah.git’. Is this expected behavior? I am watching a bunch of tutorials where people easily push to the old main/master. Why is this functionality no longer available?</p>
